controllers/rack.go

```diff
@@ -791,20 +843,34 @@ func (r *SingleClusterReconciler) scaleDownRack(
 	return found, reconcileRequeueAfter(1)
 }
 
-func (r *SingleClusterReconciler) rollingRestartRack(
-	found *appsv1.StatefulSet, rackState *RackState, ignorablePods []corev1.Pod, restartTypeMap map[string]RestartType,
-) (*appsv1.StatefulSet, reconcileResult) {
+func (r *SingleClusterReconciler) rollingRestartRack(found *appsv1.StatefulSet, rackState *RackState,
+	ignorablePods []corev1.Pod, restartTypeMap map[string]RestartType,
+	failedPods []*corev1.Pod) (*appsv1.StatefulSet, reconcileResult) {
 	r.Log.Info("Rolling restart AerospikeCluster statefulset nodes with new config")
 
 	r.Recorder.Eventf(
 		r.aeroCluster, corev1.EventTypeNormal, "RackRollingRestart",
 		"[rack-%d] Started Rolling restart", rackState.Rack.ID,
 	)
 
-	// List the pods for this aeroCluster's statefulset
-	podList, err := r.getOrderedRackPodList(rackState.Rack.ID)
-	if err != nil {
-		return found, reconcileError(fmt.Errorf("failed to list pods: %v", err))
+	var (
+		err     error
+		podList []*corev1.Pod
+	)
+
+	if len(failedPods) != 0 {
```

Comment:
nit: better to keep the condition in sync
In `upgradeRack` we are checking `if len(failedPods) == 0` here the condition is opposite